lib/tests/test_local_working_copy.rs

```diff
@@ -601,6 +601,78 @@ fn test_materialize_conflicted_files() {
     4
     >>>>>>> Conflict 1 of 1 ends
     "###);
+
+    // Editing a conflicted file should correctly propagate updates to each of
+    // the conflicting trees.
+    testutils::write_working_copy_file(
+        &workspace_root,
+        file1_path,
+        r##"<<<<<<< Conflict 1 of 1
+%%%%%%% Changes from base to side #1
+-b_edited
++a_edited
++++++++ Contents of side #2
+c_edited
+>>>>>>> Conflict 1 of 1 ends
+"##,
+    );
+
+    let edited_tree = test_workspace.snapshot().unwrap();
+    let edited_files = edited_tree
+        .entries()
+        .filter_map(|(path, value)| {
+            if *path == *file1_path {
```

Comment:
Optional: I'm not sure what's the most canonical, but `path.as_ref() == file1_path` seems slightly better to me.